src/tv2-common/helpers/rundownAdLibActions.ts

```diff
@@ -128,36 +103,67 @@ export function ParseTransitionString(transitionString: string): ActionTakeWithT
 	})
 }
 
+function makeTransitionOnTakeAction(
+	config: TV2BlueprintConfig,
+	variant: ActionTakeWithTransitionVariant,
+	transitionValues: TransitionValues
+): IBlueprintActionManifest {
+	const userData: ActionTakeWithTransition = {
+		type: AdlibActionType.TAKE_WITH_TRANSITION,
+		variant,
+		takeNow: true
+	}
+	return makeTransitionAction(config, userData, transitionValues, AdlibTags.ADLIB_TAKE_WITH_TRANSITION)
+}
+
+function makeTransitionOnNextTakeAction(
+	config: TV2BlueprintConfig,
+	variant: ActionTakeWithTransitionVariant,
+	transitionValues: TransitionValues
+): IBlueprintActionManifest {
+	const userData: ActionTakeWithTransition = {
+		type: AdlibActionType.TAKE_WITH_TRANSITION,
+		variant,
+		takeNow: false
+	}
+	return makeTransitionAction(config, userData, transitionValues, AdlibTags.ADLIB_NEXT_TAKE_WITH_TRANSITION)
+}
+
 function makeTransitionAction(
 	config: TV2BlueprintConfig,
 	userData: ActionTakeWithTransition,
-	rank: number,
-	label: string,
-	jingle: string,
-	alphaAtStart: number | undefined,
-	duration: number | undefined,
-	alphaAtEnd: number | undefined
+	transitionValues: TransitionValues,
+	adlibTag: AdlibTags
 ): IBlueprintActionManifest {
 	const tag = GetTagForTransition(userData.variant)
-	const isEffekt = !!label.match(/^\d+$/)
+	const isEffekt = !!transitionValues.label.match(/^\d+$/)
 
 	return {
-		externalId: `${JSON.stringify(userData)}_${AdlibActionType.TAKE_WITH_TRANSITION}_${rank}`,
+		externalId: `${JSON.stringify(userData)}_${AdlibActionType.TAKE_WITH_TRANSITION}_${transitionValues.rank}`,
 		actionId: AdlibActionType.TAKE_WITH_TRANSITION,
 		userData,
 		userDataManifest: {},
 		display: {
-			_rank: rank,
-			label: t(isEffekt ? `EFFEKT ${label}` : label),
+			_rank: transitionValues.rank,
+			label: t(isEffekt ? `EFFEKT ${transitionValues.label}` : transitionValues.label),
 			sourceLayerId: SharedSourceLayers.PgmAdlibJingle,
 			outputLayerId: SharedOutputLayers.PGM,
-			tags: [AdlibTags.ADLIB_STATIC_BUTTON, AdlibTags.ADLIB_TAKE_WITH_TRANSITION],
+			tags: [AdlibTags.ADLIB_STATIC_BUTTON, adlibTag],
 			currentPieceTags: [tag],
 			nextPieceTags: [tag],
 			content:
-				isEffekt || !!/^MIX ?\d+$/i.test(label) || !!/^CUT$/i.test(label) || !!/^DIP ?\d+$/i.test(label)
+				isEffekt ||
+				!!/^MIX ?\d+$/i.test(transitionValues.label) ||
+				!!/^CUT$/i.test(transitionValues.label) ||
+				!!/^DIP ?\d+$/i.test(transitionValues.label)
```

Comment:
`test` already returns a boolean.
```suggestion
				/^MIX ?\d+$/i.test(transitionValues.label) ||
				/^CUT$/i.test(transitionValues.label) ||
				/^DIP ?\d+$/i.test(transitionValues.label)
```

Reply:
Leftover from before refactor - has been updated